File: frontend/src/api/payments.ts
import { apiClient } from "../lib/http";
import type { PaymentRead, ProcessPaymentParams } from "../types/api";

export async function processPayment(orderId: string, params?: ProcessPaymentParams) {
  const response = await apiClient.post<PaymentRead>(`/payments/orders/${orderId}/process`, null, {
    params,
  });
  return response.data;
}

File: backend/app/api/v1/endpoints/payments.py
from uuid import UUID

from fastapi import APIRouter, Query

from app.api.deps import CurrentUser, DbSession
from app.schemas.payment import PaymentRead
from app.services.payment_service import PaymentService

router = APIRouter()


@router.post("/orders/{order_id}/process", response_model=PaymentRead)
async def process_payment(
    order_id: UUID,
    db: DbSession,
    _: CurrentUser,
    retry: bool = Query(default=False),
    force_outcome: str = Query(default="auto", pattern="^(auto|success|failed)$"),
) -> PaymentRead:
    service = PaymentService(db)
    payment = await service.process_mock_payment(order_id, retry=retry, force_outcome=force_outcome)
    return PaymentRead.model_validate(payment)
